lib/better_html/tokenizer/base_erb.rb

```diff
@@ -0,0 +1,78 @@
+require 'erubi'
+require_relative 'token'
+require_relative 'location'
+
+module BetterHtml
+  module Tokenizer
+    class BaseErb < ::Erubi::Engine
+      REGEXP_WITHOUT_TRIM = /<%(={1,2}|%)?(.*?)()?%>([ \t]*\r?\n)?/m
+      STMT_TRIM_MATCHER = /\A(-|#)?(.*?)(-)?\z/m
+      EXPR_TRIM_MATCHER = /\A(.*?)(-)?\z/m
+
+      attr_reader :tokens
+      attr_reader :current_position
+
+      def initialize(document)
+        @document = document
+        @tokens = []
+        @current_position = 0
+        super(document, regexp: REGEXP_WITHOUT_TRIM, trim: false)
+      end
+
+      private
+
+      def append(text)
+        @current_position += text.length
+      end
+
+      def add_code(code)
+        _, ltrim_or_debug, code, rtrim = *STMT_TRIM_MATCHER.match(code)
+        ltrim = ltrim_or_debug if ltrim_or_debug == '-'
+        indicator = ltrim_or_debug if ltrim_or_debug == '#'
+        add_erb_tokens(ltrim, indicator, code, rtrim)
+        append("<%#{ltrim}#{indicator}#{code}#{rtrim}%>")
+      end
+
+      def add_expression(indicator, code)
+        _, code, rtrim = *EXPR_TRIM_MATCHER.match(code)
+        add_erb_tokens(nil, indicator, code, rtrim)
+        append("<%#{indicator}#{code}#{rtrim}%>")
+      end
+
+      def add_erb_tokens(ltrim, indicator, code, rtrim)
+        type = case indicator
+        when nil
+          :stmt
+        when '#'
+          :debug
```

Comment:
Is there a reason to call this `:debug`? I would suggest `:comment` instead.

Reply:
You are right, I double checked and I mixed up two things:
this is a comment (replaced with spaces): https://github.com/kwatch/erubis/blob/14d3eab57fbc361312c8f3af350cbf9a5bafce17/lib/erubis/converter.rb#L148-L156
this is debug:
https://github.com/kwatch/erubis/blob/14d3eab57fbc361312c8f3af350cbf9a5bafce17/lib/erubis/converter.rb#L182-L183

but I think `===` is specific to `erubis` gem